serving-api/src/1.4/scala/com/stratio/sparta/serving/api/actor/ClusterLauncherActor.scala

```diff
@@ -0,0 +1,199 @@
+  private val ZookeeperConfig = SpartaConfig.getZookeeperConfig.get
+  private val HdfsConfig = SpartaConfig.getHdfsConfig.get
+  private val DetailConfig = SpartaConfig.getDetailConfig.get
+
+  private val Hdfs = HdfsUtils(HdfsConfig)
+  private val Uploader = ClusterSparkFiles(policy, Hdfs)
+  private val PolicyId = policy.id.get.trim
+  private val Master = ClusterConfig.getString(AppConstant.Master)
+  private val BasePath = s"/user/${Hdfs.userName}/${AppConstant.ConfigAppName}/$PolicyId"
+  private val PluginsJarsPath = s"$BasePath/${HdfsConfig.getString(AppConstant.PluginsFolder)}/"
+  private val DriverJarPath = s"$BasePath/${HdfsConfig.getString(AppConstant.ExecutionJarFolder)}/"
+
+  implicit val timeout: Timeout = Timeout(3.seconds)
+
+  override def receive: PartialFunction[Any, Unit] = {
+    case Start => doInitSpartaContext()
+  }
+
+  def doInitSpartaContext(): Unit = {
+    Try {
+      log.info("Init new cluster streamingContext with name " + policy.name)
+      validateSparkHome()
+      val driverPath = Uploader.getDriverFile(DriverJarPath)
+      val pluginsFiles = Uploader.getPluginsFiles(PluginsJarsPath)
+      val driverParams = Seq(PolicyId, zkConfigEncoded, detailConfigEncoded, pluginsEncoded(pluginsFiles))
+
+      launch(SpartaDriver, driverPath, Master, sparkArgs, driverParams, pluginsFiles)
+    } match {
+      case Failure(exception) =>
+        log.error(exception.getLocalizedMessage, exception)
+        setErrorStatus()
+      case Success(_) =>
+      //TODO add more statuses for the policies
+    }
+  }
+
+  private def setErrorStatus(): Unit =
+    policyStatusActor ? Update(PolicyStatusModel(policy.id.get, PolicyStatusEnum.Failed))
+
+  private def sparkHome: String = Properties.envOrElse("SPARK_HOME", ClusterConfig.getString(AppConstant.SparkHome))
+
+  /**
+   * Checks if we have a valid Spark home.
+   */
+  private def validateSparkHome(): Unit = require(Try(sparkHome).isSuccess,
+    "You must set the $SPARK_HOME path in configuration or environment")
+
+  /**
+   * Checks if supervise param is set when execution mode is standalone
+   *
+   * @return The result of checks as boolean value
+   */
+  def isStandaloneSupervise: Boolean =
+    if (DetailConfig.getString(AppConstant.ExecutionMode) == AppConstant.ConfigStandAlone) {
+      Try(ClusterConfig.getBoolean(AppConstant.StandAloneSupervise)).getOrElse(false)
+    } else false
+
+  private def launch(main: String, hdfsDriverFile: String, master: String, args: Map[String, String],
+                     driverParams: Seq[String],
+                     pluginsFiles: Seq[String]): Unit = {
+    val sparkLauncher = new SparkLauncher()
+      .setSparkHome(sparkHome)
+      .setAppResource(hdfsDriverFile)
+      .setMainClass(main)
+      .setMaster(master)
+    pluginsFiles.foreach(file => sparkLauncher.addJar(file))
+    args.map({ case (k: String, v: String) => sparkLauncher.addAppArgs(k, v) })
```

Comment:
+1, I see that both classes are practically equal. Maybe it could be a good idea use heritage and override the appropriate method